### ISSUE TITLE
`Slider` and `RangeSlider` thumb accessibility issues

### ISSUE THREAD

**Issue by TahaTesser** [flutter/flutter#114225]
### Description
#### 1. The entire `Slider` is focused instead of just the thumb. 

The focus window should only follow the thumb while keeping the ability to tap anywhere on the slider to change value.
(this especially becomes problematic with `RangeSlider`)

| Flutter | Android |
| --------------- | --------------- |
| <img src="https://user-images.githubusercontent.com/48603081/198578617-56b85c4d-4f43-4eed-9c80-0d9b8f7ccbd5.jpg" /> | <img src="https://user-images.githubusercontent.com/48603081/198578513-f64acb71-8c5b-482a-a866-857d68d81084.jpg" /> |




#### 2. `RangeSider` focus window is divided in half despite range thumb locations instead of just the range thumbs.

1. Focus window takes half the size regardless of the thumb location
2. If both range thumbs are located on one side of the slider, the focus window is still shown on the otherwise side, in an empty slider area.
3. Focus windows should be on the range thumbs





| Flutter | Android |
| --------------- | --------------- |
| <img src="https://user-images.githubusercontent.com/48603081/198580896-39585afc-d587-4488-8549-b250d3266522.jpg" /> | <img src="https://user-images.githubusercontent.com/48603081/198580902-fa0442c7-8b00-495e-b4d5-35c1679b6948.jpg" /> |
| <img src="https://user-images.githubusercontent.com/48603081/198580960-c67da866-3fef-42a2-9263-8bb9149d70a2.jpg" /> | <img src="https://user-images.githubusercontent.com/48603081/198580968-b4f1dc50-4a43-44e5-b0a7-55322d81a34e.jpg" /> |

**Comment by TahaTesser** [comment]
This is needed for https://github.com/flutter/flutter/issues/53958

**Comment by TahaTesser** [comment]
cc: @Piinks

**Comment by Piinks** [comment]
M3 has not given us an a11y spec yet (https://m3.material.io/components/sliders/accessibility), so I would follow the spec that preceded it from M2 https://m2.material.io/components/sliders, which is also lacking specific a11y behavior, but it looks like you have this well defined based on the Android behavior.

cc @chunhtai & @gspencergoog if there are any other a11y or focus issues.

Thanks for identifying this!

**Comment by chunhtai** [comment]
Since there is a native android equivalence in this case, we should follow OEM behavior.

**Comment by TahaTesser** [comment]
https://github.com/flutter/flutter/pull/115285 is reverted

**Comment by TahaTesser** [comment]
cc: @pdblasi-google @Piinks 

https://github.com/flutter/flutter/pull/115285 was reverted for unknown reasons. 

Should I file a new PR so we run the tests and investigate the cause?

**Comment by hannah-hyj** [comment]
I will try to reland https://github.com/flutter/flutter/pull/121598 again, i think the test failures was because of semantics tree structure changes, thumb become the leaf node and overlay node has their own nodes, i will investigate the google 3 failures
